### PR TITLE
radmin: Update to use fr_time_delta_t

### PR DESCRIPTION
**Pull request by jpereira**

### CODE REVIEW

File: src/lib/util/print.c [hunk 1]
```diff
@@ -838,3 +838,38 @@ char *fr_asprintf(TALLOC_CTX *ctx, char const *fmt, ...)
...
+ *
+ * @param[in] ctx	to allocate buffer in.
+ * @param[in] fp	to write the result of fmt string.
+ * @param[in] fmt	string.
+ * @param[in] ...	variadic argument list.
+ * @return
+ *   - On success, the number of bytes written is returned (zero indicates nothing was written).
+ *   - On error, -1 is returned, and errno is set appropriately
+ */
+int fr_fprintf(TALLOC_CTX *ctx, FILE *fp, char const *fmt, ...)
```
Comment (arr2036) [inline src/lib/util/print.c]: This doesn't need to take a talloc ctx.  If you want it to be efficient, you should allocate a thread specific memory pool and use that instead... But I think fr_vasprintf already does that for the intermediary buffers, so you won't save  that much CPU time.

**Comment by arr2036** [comment]
Looks fine other than that one issue

**Comment by jpereira** [comment]
@arr2036 done!

File: src/bin/radmin.c [hunk 2]
```diff
@@ -452,16 +452,11 @@ static int cmd_terminate(UNUSED FILE *fp, UNUSED FILE *fp_err, UNUSED void *ctx,
 
 static int cmd_uptime(FILE *fp, UNUSED FILE *fp_err, UNUSED void *ctx, UNUSED fr_cmd_info_t const *info)
 {
-	struct timeval now;
+	fr_time_t now;
```
Comment (arr2036) [inline src/bin/radmin.c]: ah and that's an fr_time_delta_t not an fr_time_t, and maybe rename it to uptime.

**Comment by jpereira** [comment]
@arr2036 done

File: src/bin/radmin.c [hunk 3]
```diff
@@ -452,16 +452,11 @@ static int cmd_terminate(UNUSED FILE *fp, UNUSED FILE *fp_err, UNUSED void *ctx,
...
-	now = fr_time_to_timeval(fr_time());
-	fr_timeval_subtract(&now, &now, &start_time);
+	uptime = fr_time() - start_time;
 
-	fprintf(fp, "Uptime: %u.%06u seconds\n",
-		(unsigned int) now.tv_sec,
-		(unsigned int) now.tv_usec);
-
-	return 0;
+	return fr_fprintf(fp, "Uptime: %pVs seconds\n", fr_box_time_delta(uptime));
```
Comment (arr2036) [inline src/bin/radmin.c]: This previously returned 0, now it'll return the number of bytes written?

File: src/lib/util/print.c [hunk 4]
```diff
@@ -838,3 +838,38 @@ char *fr_asprintf(TALLOC_CTX *ctx, char const *fmt, ...)
...
+ *
+ * @param[in] ctx	to allocate buffer in.
+ * @param[in] fp	to write the result of fmt string.
+ * @param[in] fmt	string.
+ * @param[in] ...	variadic argument list.
+ * @return
+ *   - On success, the number of bytes written is returned (zero indicates nothing was written).
+ *   - On error, -1 is returned, and errno is set appropriately
+ */
+int fr_fprintf(FILE *fp, char const *fmt, ...)
```
Comment (arr2036) [inline src/lib/util/print.c]: should be a ssize_t

File: src/lib/util/print.c [hunk 5]
```diff
@@ -838,3 +838,38 @@ char *fr_asprintf(TALLOC_CTX *ctx, char const *fmt, ...)
...
+	if (fd < 0) {
+		fr_strerror_printf("Invalid 'fp'");
+		return -1;
+	}
+
+	va_start(ap, fmt);
+	buf = fr_vasprintf(NULL, fmt, ap);
+	va_end(ap);
+
+	ret = (int)write(fd, buf, strlen(buf));
```
Comment (arr2036) [inline src/lib/util/print.c]: No need to cast

**Pull request merged**